### PR TITLE
Go / Helm vulnerabilities release-1.15

### PR DESCRIPTION
**Pull request by nmnellis**
**Please provide a description of this PR:**

Updated libraries to remove go / helm vulnerabilities

Fixes: CVE-2022-41717, CVE-2022-23524, CVE-2022-23525, CVE-2022-23526

Issue: https://github.com/istio/istio/issues/42583 

Related: https://github.com/istio/istio/pull/42584

### CODE REVIEW

**Review by howardjohn — CHANGES_REQUESTED** [review]
I'm not very comfortable with these nontrivial updates to fix CVEs that are not in any way impacting Istio users. It's one thing to make low risk updates to appease scanners, but bringing real risk to users for not *functional* improvements needs further discussion IMO.

To be clear - this only really applies to this one, the 1.16 PR is probably fine.

**Comment by nmnellis** [comment]
> I'm not very comfortable with these nontrivial updates to fix CVEs that are not in any way impacting Istio users. It's one thing to make low risk updates to appease scanners, but bringing real risk to users for not _functional_ improvements needs further discussion IMO.

i dont disagree. this was picked up in a customer scanner as well

File: releasenotes/notes/42583.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: bug-fix
+area: security
+
+# issue is a list of GitHub issues resolved in this note.
+issue:
+  - 42583
+
+releaseNotes:
+  - |
+    **Fixed** Updated golang/helm libraries to fix CVE-2022-41717, CVE-2022-23524, CVE-2022-23525, CVE-2022-23526.
```
Comment (ericvn) [inline releasenotes/notes/42583.yaml:11]: Fixed updated is a double Action. You could probably just go with **Updated**

**Comment by dhawton** [comment]
Discussed in WG meeting: The change was not backported to Helm 3.9 and there are concerns over the minor version dependency bumps between patch versions. The CVE does not appear to affect Istio, so we're not going to pursue this change at this time.

**Comment by nrjpoddar** [comment]
Should we be patching the https://github.com/advisories/GHSA-xrjj-mj9h-534m one if not the Helm ones?

**Pull request closed without merging**